transform/mattermost-analytics/models/intermediate/product/licenses/_licenses_servers/int_self_hosted_licenses.sql

```diff
@@ -0,0 +1,33 @@
+with cws_licenses as (
+    select distinct cl.license_id
+        , cl.customer_id as customer_id
+        , cl.customer_email as email
+        , cl.company_name as company_name
+        , coalesce (p.name, cl.sku_short_name) as license_name
+        , 'CWS' as source
+    from {{ ref('stg_cws__license') }} cl
+    left join {{ ref('stg_stripe__products')}} p on cl.stripe_product_id = p.product_id
+), stripe_licenses as (
+    select distinct license_id
+        , c.portal_customer_id as customer_id
+        , c.email as email
+        , c.name as company_name
+        , coalesce(s.edition, p.name) as license_name
+        , 'Stripe' as source
+    from {{ ref('stg_stripe__subscriptions')}}  s 
+    join {{ ref('stg_stripe__customers')}}  c on s.customer_id = c.customer_id
+    left join {{ ref('stg_stripe__products')}} p on coalesce(s.product_id, s.current_product_id) = p.product_id
+    where license_id is not null
```

Comment:
why this check is only here and not for the other two?

Reply:
CWS doesn't contain cloud licenses. I validated this and presented findings in one of our meetings.
Legacy source data, doesn't contain `installation_id`, so a check there wasn't required.

Reply:
this is for self-hosted `license_id is not null`

Reply:
yes, the others don't contain any cloud data.

Stripe contains both cloud and self-hosted data hence the `license_id` filter, the check for cloud using `installation_id` field.

Reply:
ok, so for self-hosted licenses it is ok to have null `license_ids` in the result set of the intermediate model - except for Stripe?

Reply:
However, this is generating duplicate data. License information can be present in multiple sources causing duplicates when we join with server data. I'm trying to find a way to assign an order of precedence